### DISCUSSION TITLE
Serializing catalog items to Json in V8

### DISCUSSION THREAD

**Discussion by himby** [TerriaJS/terriajs#6208] [Q&A] [answered]
Hi,
In v7 I utilized a serializeToJson function on the catalog items to convert them to a format equal to the catalog items in the initialization files. I'm trying to do something similar in v8, but this function doesn't work for the purpose any more. I need a feature that can convert any type of catalog item. I see that BuildShareLink.js does have some mechanisms that could be used. but this seems unfinished. Any tips to achieve this?

**Comment by tephenavies** [comment, marked as answer]
Hi @himby. I think the following may work:
```ts
const item: BaseModel;
Object.keys(item.traits).forEach(traitId => {
    const trait = item.traits[traitId];
    const value = item[traitId];
    const jsonValue = trait.toJson(value);
    if (jsonValue !== undefined) {
      result[traitId] = jsonValue;
    }
  });
```
though you will get many entries in that JSON that come through from Trait defaults.

Built from [saveStratumToJson](https://github.com/TerriaJS/terriajs/blob/57608473d69abd511ed804e5e79b8c3da6c852b5/lib/Models/Definition/saveStratumToJson.ts#L5)

**Reply by himby** [reply]
Thanks @steve9164 ! That tip sent me in the right direction. Very helpful!
